src/store/app/app.slice.ts

```diff
@@ -51,6 +55,73 @@ export const appSlice = createSlice({
     ) => {
       state.walletNotConnectedModal = payload;
     },
+    setLocalXusdTransactions: (
+      state,
+      { payload }: PayloadAction<XusdLocalTransaction>
+    ) => {
+      if (!state.chainId || !state.account) {
+        return;
+      }
+
+      if (!state.xusdLocalTransactions[state.chainId]) {
+        state.xusdLocalTransactions[state.chainId] = {
+          [state.account]: [payload],
+        };
+        return;
+      }
+
+      if (!state.xusdLocalTransactions[state.chainId][state.account]) {
+        state.xusdLocalTransactions[state.chainId][state.account] = [payload];
+        return;
+      }
+
+      state.xusdLocalTransactions[state.chainId][state.account].unshift(
+        payload
+      );
+    },
+    updateLocalXusdTransactionStatus: (
+      state,
+      { payload }: PayloadAction<UpdateTxStatus>
+    ) => {
+      if (
+        !state.chainId ||
+        !state.account ||
+        !state.xusdLocalTransactions[state.chainId] ||
+        !state.xusdLocalTransactions[state.chainId][state.account]
+      ) {
+        return;
+      }
+
+      const txToUpdate = state.xusdLocalTransactions[state.chainId][
+        state.account
+      ].find(({ txHash }) => txHash === payload.txHash);
+
+      if (!txToUpdate || txToUpdate.status === payload.newStatus) {
+        return;
+      }
+
+      txToUpdate.status = payload.newStatus;
+    },
+    removeLocalXusdTransactions: (
+      state,
+      { payload }: PayloadAction<XusdLocalTransaction[]>
+    ) => {
+      if (
```

Comment:
I think the same check is being done in 3 places: 'removeLocalXusdTransactions', `updateLocalXusdTransactionStatus` and the `xusdLocalTransactionsSelector`. You could extract it to a function